File: backend/master/src/main/java/com/ociet/loadgenerator/master/Controller.java
package com.ociet.loadgenerator.master;

import com.netflix.discovery.shared.Application;
import com.netflix.eureka.EurekaServerContextHolder;
import com.ociet.loadgenerator.master.dto.LoadRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@CrossOrigin(origins = "*")
public class Controller {
    private final RequestService requestService;

    @Autowired
    Controller(RequestService requestService) {
        this.requestService = requestService;
    }

    @PostMapping(path="/load")
    public void requestLoad(@RequestBody LoadRequest loadRequest) {
        requestService.requestLoad(loadRequest);
    }

    @GetMapping(path="/slaves")
    public int getSlaveCount() {
        Application application = EurekaServerContextHolder.getInstance().getServerContext().getRegistry().getApplication("SLAVE2");
        return application == null ? 0 : application.size();
    }
}

File: backend/master/src/main/kotlin/com/chylb/loadgenerator/master/Controller.kt
package com.chylb.loadgenerator.master

import com.chylb.loadgenerator.master.dto.LoadRequest
import com.netflix.discovery.shared.Application
import com.netflix.eureka.EurekaServerContextHolder
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*


@RestController
@CrossOrigin(origins = ["*"])
class Controller @Autowired internal constructor(requestService: RequestService) {
    private val requestService: RequestService
    @PostMapping(path = ["/load"])
    fun requestLoad(@RequestBody loadRequest: LoadRequest?) {
        if (loadRequest != null) {
            requestService.requestLoad(loadRequest)
        }
    }

    @GetMapping(path = ["/slaves"])
    fun slaveCount(): Int {
        val application: Application? =
            EurekaServerContextHolder.getInstance().serverContext.registry.getApplication("SLAVE2")
        return if (application == null) 0 else application.size()
    }

    init {
        this.requestService = requestService
    }
}